build/build-css-class-list.js

```diff
@@ -0,0 +1,109 @@
+/**
+ * @license Copyright 2021 The Lighthouse Authors. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License. You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
+ */
+
+'use strict';
+
+const fs = require('fs');
+const CssSelectorParser = require('css-selector-parser').CssSelectorParser;
+const jsdom = require('jsdom');
+const {LH_ROOT} = require('../root.js');
+
+// Read report CSS from a sample-report and also from templates.html (as JSDOM won't execute the JS that'd inject their styles)
+const sampleStandaloneHtml = fs.readFileSync(`${LH_ROOT}/dist/now/english/index.html`, 'utf-8');
+const templatesHtml = fs.readFileSync(`${LH_ROOT}/report/assets/templates.html`, 'utf-8');
+
+/**
+ * First step, extract selectors from stylesheets
+ * @param {string} html
+ */
+function getSelectorsFromStyleSheets(html) {
+  const {window} = new jsdom.JSDOM(html);
+
+  // unwrap any <template> tags
+  Array.from(window.document.querySelectorAll('template')).forEach(templateElem => {
+    window.document.body.append(templateElem.content.cloneNode(true));
+  });
+
+  const selectors = Array.from(window.document.styleSheets)
+    .flatMap(sheet => Array.from(sheet.cssRules || []))
+    .map(rule => /** @type {CSSStyleRule} */ (rule).selectorText)
+    .filter(Boolean);
+  return selectors;
+}
+const foundSelectors = [];
+foundSelectors.push(...getSelectorsFromStyleSheets(sampleStandaloneHtml));
+foundSelectors.push(...getSelectorsFromStyleSheets(templatesHtml));
+const uniqSelectors = Array.from(new Set(foundSelectors));
+
+// Shout out to alecxe for the pattern. https://stackoverflow.com/a/38422908/89484
```

Comment:
```suggestion
// https://stackoverflow.com/a/38422908/89484
```

nit ... as a reader, I just want the link :)

Reply:
why ya gotta do alecxe dirty like that @connorjclark 😉

Reply:
hehe... of course, I have nothing against giving attribution, but that should be done with [`Co-authored-by`](https://docs.github.com/en/github/committing-changes-to-your-project/creating-and-editing-commits/creating-a-commit-with-multiple-authors)